contrib/verify-commits/verify-commits.py

```
@@ -148,15 +148,22 @@ def main():
         allow_unclean = current_commit in unclean_merge_allowed
         if len(parents) == 2 and check_merge and not allow_unclean:
             current_tree = subprocess.check_output([GIT, 'show', '--format=%T', current_commit]).decode('utf8').splitlines()[0]
-            subprocess.call([GIT, 'checkout', '--force', '--quiet', parents[0]])
-            subprocess.call([GIT, 'merge', '--no-ff', '--quiet', '--no-gpg-sign', parents[1]], stdout=subprocess.DEVNULL)
-            recreated_tree = subprocess.check_output([GIT, 'show', '--format=format:%T', 'HEAD']).decode('utf8').splitlines()[0]
+            # This merge-tree functionality requires git >= 2.38. The
+            # --write-tree option was added in order to opt-in to the new
+            # behavior. Older versions of git will not recognize the option and
+            # will instead exit with code 128.
+            try:
+                recreated_tree = subprocess.check_output([GIT, "merge-tree", "--write-tree", parents[0], parents[1]]).decode('utf8').splitlines()[0]
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)